pkg/testing/fetcher_local.go

```diff
@@ -29,9 +30,10 @@ func WithLocalSnapshotOnly() localFetcherOpt {
 }
 
 // LocalFetcher returns a fetcher that pulls the binary of the Elastic Agent from a local location.
-func LocalFetcher(dir string, opts ...localFetcherOpt) Fetcher {
+func LocalFetcher(dir string, binary string, opts ...localFetcherOpt) Fetcher {
```

Comment:
This is weird to have as a top-level option. Being that this is mainly used to test Elastic Agent please move this to an option just like the usage of snapshot first is.

I would expect the usage to be:

```
testing.LocalFetcher(".", testing.WithBinaryName(binary))
```